include/ocpp/v201/smart_charging.hpp

```diff
@@ -48,29 +56,51 @@ std::ostream& operator<<(std::ostream& os, const ProfileValidationResultEnum val
 class SmartChargingHandler {
 private:
     std::map<int32_t, std::unique_ptr<EvseInterface>>& evses;
+    std::shared_ptr<DeviceModel>& device_model;
 
     std::shared_ptr<ocpp::v201::DatabaseHandler> database_handler;
     // cppcheck-suppress unusedStructMember
     std::map<int32_t, std::vector<ChargingProfile>> charging_profiles;
     std::vector<ChargingProfile> station_wide_charging_profiles;
 
 public:
-    explicit SmartChargingHandler(std::map<int32_t, std::unique_ptr<EvseInterface>>& evses);
+    explicit SmartChargingHandler(std::map<int32_t, std::unique_ptr<EvseInterface>>& evses,
+                                  std::shared_ptr<DeviceModel>& device_model);
 
+    ///
+    /// \brief validates the given \p profile according to the specification.
+    /// If a profile does not have validFrom or validTo set, we conform the values
+    /// to a representation that fits the spec.
+    ///
+    ProfileValidationResultEnum validate_profile(ChargingProfile& profile, int32_t evse_id);
```

Comment:
They have to:

> If a profile does not have validFrom or validTo set, we conform the values to a representation that fits the spec.

And in `validate_profile_schedules()`:

> If a profiles charging schedule period does not have a valid numberPhases, we set it to the default value (3).

Reply:
I have also noted this issue before and still agree with @shankari here. I think it should be possible to separate the responsibilities a bit. Maybe we can already update those fields before calling the validate functions? Would make it a lot clearer what is going on, something along the lines of: `set_default_number_of_phases()` and `set_default_valid_from_to_times()`

Reply:
We understand the idea of using `const`, but this allows us to set the defaults required by the specification while keeping the public footprint minimal. As-is, `validate_profile()` and `add_profile()` can be called directly without needing to know about the internals of the profiles.

Reply:
Of course I understand that you do ;) I just think this is not ideal. On the other hand. it would also be a small update down the line if we ever decide to change this so I won't let this block the PR.

Reply:
I agree with @marcemmers that while this is not important enough to hold up the PR (and I approved it as well), I would like to have this tracked as a cleanup change in a subsequent PR. When I read `validate_XXX`, I don't expect that it will modify the input parameters.